### DISCUSSION TITLE
Disable Next.js routing

### DISCUSSION THREAD

**Discussion by lmatteis** [vercel/next.js#21976] [Help]
We built a Next app and it's published on http://foo.com/whatever/page.

Another app (http://bar.com) consumes this app by fetching the above resource, obtaining the HTML and somehow appending it to the DOM via javascript. 

When this happens bar.com throws an exception: `Failed to execute 'replaceState' on 'History' ` because apparently the Next app on foo.com tries to replace history and control the routing (which it is not allowed to do):

![image](https://user-images.githubusercontent.com/535442/107284352-9769f480-6a55-11eb-9172-ebe4411fab9d.png)

Is there a way to disable routing for a Next app so it doesn't try to replaceState? We don't need routing as we simply need to load a large landing page.

**Comment by jamesmosier** [comment]
Hi @lmatteis. You can use the [`useFileSystemPublicRoutes: false`](https://nextjs.org/docs/advanced-features/custom-server#disabling-file-system-routing) option in your `next.config.js` to disable Next's routing capabilities.

**Reply by lmatteis** [reply]
Thanks @jamesmosier but I've tried both that and `router.beforePopState(() => false)` which was suggested from the link and I keep on getting the same error.
